src/EventSubscriber/CartEventSubscriber.php

```diff
@@ -33,42 +33,11 @@ public function __construct(PrivateTempStoreFactory $tempstore_private) {
    * {@inheritdoc}
    */
   public static function getSubscribedEvents() {
-    $events[CartEvents::CART_ENTITY_ADD][] = ['addOptions'];
     $events[CartEvents::CART_ENTITY_ADD][] = ['mergeOrderItems'];
 
     return $events;
   }
 
-  /**
-   * Adds the option values to the new order item.
-   *
-   * @param \Symfony\Component\EventDispatcher\Event $event
-   *   The dispatched event.
-   */
-  public function addOptions(Event $event) {
```

Comment:
@safjoseph So, with removing this, does merging of the cart items still continue to work depending on the selected options? Will the cart items will automatically increment the quantities (instead of adding a new item to the cart) if the user selected a product with a specific option and add them to cart, then, comes back and selects the same product with the same option again?

Reply:
Yep. It works as expected.